src/include/sof/schedule/ll_schedule.h

```diff
@@ -34,7 +34,7 @@ struct ll_task_pdata {
 	uint16_t skip_cnt;	/**< how many times the task was skipped for execution */
 };
 
-#if !defined(__ZEPHYR__) || defined(CONFIG_IMX)
+#if !defined(__ZEPHYR__) || (defined(CONFIG_IMX) && !CONFIG_PREEMPT_DOMAIN)
 int scheduler_init_ll(struct ll_schedule_domain *domain);
 
```

Comment:
why not use `!defined(CONFIG_PREEMPT_DOMAIN)` ?

Reply:
because CONFIG_PREEMPT_DOMAIN is technically defined when on Zephyr (please see zephyr/Kconfig) so we need to actually check if it was set to y or no